### PR TITLE
use more strict formatting

### PR DESCRIPTION
**Pull request by wilsonwang371**
## Why are these changes needed?

Use a more strict format check with gofumpt

We discussed this a long time before. Due to some conflicts with our internal branch, we postponed this until now. Since our downstream branch and upstream branch is in sync now, it is a good time for us to apply a more strict code formatting on kuberay.

## Related issue number

N/A

## Checks

- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [X] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wilsonwang371** [comment]
@scarlet25151

**Review by DmitriGekhtman — APPROVED** [review]
Great! The less we have to think about format, the better :)

**Comment by DmitriGekhtman** [comment]
@wilsonwang371 
You might want to give the collab slack channel a heads up, just in case.

File: .github/workflows/test-job.yaml [hunk 1]
```diff
@@ -88,6 +88,18 @@ jobs:
...
       if: failure()
 
+    - name: Run gofumpt
+      run: |
+        go install mvdan.cc/gofumpt@latest
+        test -z "$(set -o pipefail && $(go env GOPATH)/bin/gofumpt -l apiserver/ ray-operator/ cli/ | tee gofumpt.out)" || { cat gofumpt.out && exit 1; }
+
+    - name: Open this to see how to fix gofumpt if it fails
+      run: |
+        echo "Run command 'gofumpt -w apiserver/ ray-operator/ cli/' to correct your code format."
```
[the remark this answers is not in the document]
Comment (scarlet25151) [reply .github/workflows/test-job.yaml:98]: got it, it seems like some codes writing practice issues and gofumpt may not correct them, in this case we may need to modify manually.

**Review by scarlet25151 — APPROVED** [review]
LGTM.

**Comment by Jeffwan** [comment]
@wilsonwang371 Let's rebase the change and we can unblock the PR.

**Review by Jeffwan — APPROVED** [review]

**Comment by Jeffwan** [comment]
The change looks good to me. @DmitriGekhtman any more feedbacks?

**Pull request merged**